app/frontend/src/components/AvatarInput.test.tsx

```diff
@@ -0,0 +1,175 @@
+    expect(screen.getByAltText(getAvatarLabel(NAME))).toBeVisible();
+    expect(screen.getByAltText(getAvatarLabel(NAME))).toHaveProperty(
+      "src",
+      MOCK_INITIAL_SRC
+    );
+  });
+
+  it("uploads upon confirmation and submits key", async () => {
+    userEvent.upload(screen.getByLabelText(SELECT_AN_IMAGE), MOCK_FILE);
+
+    await waitFor(() => {
+      expect(screen.getByLabelText(CONFIRM_UPLOAD)).toBeVisible();
+      expect(screen.getByLabelText(CANCEL_UPLOAD)).toBeVisible();
+    });
+
+    userEvent.click(screen.getByLabelText(CONFIRM_UPLOAD));
+
+    await waitFor(() => {
+      expect(uploadFileMock).toHaveBeenCalled();
+    });
+
+    userEvent.click(screen.getByRole("button", { name: SUBMIT }));
+
+    await waitFor(() => {
+      expect(submitForm).toHaveBeenCalledWith({ avatarInput: MOCK_KEY });
+    });
+  });
+
+  it("cancels when cancel button pressed and doesn't submit key", async () => {
+    userEvent.upload(screen.getByLabelText(SELECT_AN_IMAGE), MOCK_FILE);
+
+    await waitFor(() => {
+      expect(screen.getByLabelText(CONFIRM_UPLOAD)).toBeVisible();
+      expect(screen.getByLabelText(CANCEL_UPLOAD)).toBeVisible();
+    });
+
+    userEvent.click(screen.getByLabelText(CANCEL_UPLOAD));
+
+    await waitFor(() => {
+      expect(uploadFileMock).not.toHaveBeenCalled();
+    });
+
+    userEvent.click(screen.getByRole("button", { name: SUBMIT }));
+
+    await waitFor(() => {
+      expect(submitForm).toHaveBeenCalledWith({ avatarInput: "" });
+    });
+    expect(screen.getByAltText(getAvatarLabel(NAME))).toHaveProperty(
+      "src",
+      MOCK_INITIAL_SRC
+    );
+  });
+
+  it(`uploading, confirming, then uploading again, then cancelling, goes back to the first upload instead of the original`, async () => {
+    const OTHER_MOCK_FILE = new File([], "firstImage.jpg");
+    uploadFileMock.mockResolvedValueOnce({
+      file: OTHER_MOCK_FILE,
+      filename: OTHER_MOCK_FILE.name,
+      key: "firstKey",
+      thumbnail_url: "thumb0.jpg",
+      full_url: "full0.jpg",
+    });
+
+    userEvent.upload(screen.getByLabelText(SELECT_AN_IMAGE), OTHER_MOCK_FILE);
+
+    await waitFor(() => {
+      expect(screen.getByLabelText(CONFIRM_UPLOAD)).toBeVisible();
+    });
+
+    userEvent.click(screen.getByLabelText(CONFIRM_UPLOAD));
+
+    await waitFor(() => {
+      expect(uploadFileMock).toHaveBeenCalled();
+    });
+
+    expect(screen.getByAltText(getAvatarLabel(NAME))).not.toHaveProperty(
+      "src",
+      MOCK_INITIAL_SRC
+    );
+
+    userEvent.upload(screen.getByLabelText(SELECT_AN_IMAGE), MOCK_FILE);
+
+    await waitFor(() => {
+      expect(screen.getByLabelText(CONFIRM_UPLOAD)).toBeVisible();
+    });
+
+    userEvent.click(screen.getByLabelText(CONFIRM_UPLOAD));
+
+    await waitFor(() => {
+      expect(uploadFileMock).toHaveBeenCalledTimes(2);
+    });
+
+    expect(screen.getByAltText(getAvatarLabel(NAME))).not.toHaveProperty(
+      "src",
+      MOCK_INITIAL_SRC
+    );
+
```

Comment:
```suggestion
    expect(screen.getByAltText(getAvatarLabel(NAME))).toHaveProperty(
      "src",
      "thumb0.jpg"
    );
```
Checking it has the src of the first upload implies it's not `MOCK_INITIAL_SRC`